bin/main

```diff
@@ -1,70 +1,121 @@
+  valid = false
+  until valid
+    if arr[player_choice] == 'X' || arr[player_choice] == 'O'
+      puts 'Wrong postion, please choose another one'
+      return valid
+    else
+      arr[player_choice] = player.token
+      valid = true
+      break
+    end
+  end
+end
+
+def self.reset
+  options = %w[y n yes no]
+
+  loop do
+    puts 'Do you want to play again?: (y/n)'
+    player_option = gets.chomp.downcase
+    if options.include? player_option
+      if player_option == 'yes' || player_option == 'y'
+        check = true
+      else
+        check = false
+      end
+      return check
+    else
+      puts 'Sorry, wrong option please enter a valid option.'
+    end
+    check
+  end
+end
+
 puts " Hello and welcome to Tic Tac Toe Game."
-# Ask the first player for nikename
 puts "First player enter your nickname: "
-# save player one name
-player_one = gets.chomp
-# print out player one name
-puts "First player name: #{player_one} "
-# Ask the second player nickname
+name_one = gets.chomp
+player_one = Player.new(name_one, 'X')
+puts "First player name: #{name_one}, your token is 'X' "
 puts "Second player enter your nickname: "
-# save player two nickname
-player_two = gets.chomp
-# print out
-puts "Second player name: #{player_two} "
-# Election of tokens
-puts "First player choose your Token, write: X or O"
-item_one = gets.chomp
-item_two = "O"
-# print out the tokens for the players
-puts "Player one has #{item_one} and Player two has #{item_two}"
-# Game start
-puts "Let's start"
-# game_ends = if there's a winner or a draw
-# while !game_ends
-# Present the table and the positons to the users
-puts "{[0],[1],[2],
-   [3],[4],[5],
-   [6],[7],[8]}"
-# Create a empty array with nine positions
-table = [[], [], [], [], [], [], [], [], []]
-# Ask to the players to choose the first position of his token
-win = false
-until win
-  loop do
-    # asking move for each player
-    puts "#{player_one}, now it\s your move, choose a number from the available boxs on the board: "
-    entry = gets.chomp
-    puts "Your #{item_one} is in: #{table.each_with_index(entry)}"
-    # displaying the board
-    puts "{[0],[1],[2],
-    [3],[X],[5],
-    [6],[7],[8]}"
-    # asking move for each player
-    puts "#{player_two}, now it\s your move, choose a number from the available boxs on the board: "
-    entry_two = gets.chomp
-    # print the position that they choose. example!
-    # if player_two == true move
-    puts "Your #{item_one} is in: #{table.each_with_index(entry_two)}"
-    # displaying the board
-    puts "{[0],[1],[2],
-      [3],[X],[5],
-      [6],[7],[O]}"
-    # else
-    puts "Error: invalid move choose another box from the board"
-    # end
+name_two = gets.chomp
+player_two = Player.new(name_two, 'O')
+puts "Second player name: #{name_two}, your token is 'O' "
+
+win_lose = false
+until win_lose
+  arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+  board = Board.new
+  puts board.display_board(arr)
```

Comment:
That line literally means: **I display a Board object**. For that, the array is not important, literally.
```rb
board.display_board()
```
Is the good way. Into that method, you will display the array mentioned previously.

Reply:
:100: